iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/PipeConsensusServerImpl.java

```diff
@@ -274,21 +282,45 @@ public synchronized void checkConsensusPipe(Map<ConsensusPipeName, PipeStatus> e
 
   public TSStatus write(IConsensusRequest request) {
     try {
+      long consensusWriteStartTime = System.nanoTime();
       stateMachineLock.lock();
+      long getStateMachineLockTime = System.nanoTime();
+      // statistic the time of acquiring stateMachine lock
+      pipeConsensusServerMetrics.recordGetStateMachineLockTime(
+          getStateMachineLockTime - consensusWriteStartTime);
       if (request instanceof ComparableConsensusRequest) {
         ((ComparableConsensusRequest) request)
             .setProgressIndex(progressIndexManager.assignProgressIndex(thisNode.getGroupId()));
       }
-      return stateMachine.write(request);
+
+      long writeToStateMachineStartTime = System.nanoTime();
+      TSStatus result = stateMachine.write(request);
+      long writeToStateMachineEndTime = System.nanoTime();
+      // statistic the time of writing request into stateMachine
+      pipeConsensusServerMetrics.recordWriteStateMachineTime(
```

Comment:
Consider "PERFORMANCE_OVERVIEW_METRICS", like in IoTConsensusServerImpl.

Reply:
Good catch! fixed